scrapy/settings/__init__.py

```diff
@@ -102,10 +111,18 @@ def get(self, name, default=None):
 
         :param default: the value to return if no setting is found
         :type default: any
+
+        :param required: if ``True`` and `default` is not specified, it will
+            raise an :class:`~scrapy.exceptions.NotConfigured` exception if the
+            specified setting has not been defined
+        :type required: boolean
         """
-        return self[name] if self[name] is not None else default
+        value = default if name not in self.attributes else self[name]
+        if required and value is _NOT_SET:
+            raise NotConfigured
```

Comment:
Should we indicate here the name of the setting in question?

Reply:
I think that could be handy, maybe:

```python
raise NotConfigured('Please configure {} in the project settings'.format(name))
```